src/main/java/org/skife/jdbi/v2/unstable/BindIn.java

```diff
@@ -90,21 +100,98 @@ public void apply(SQLStatement q) throws SQLException
     class BindingFactory implements BinderFactory<BindIn>
     {
         @Override
-        public Binder build(BindIn in)
+        public Binder build(final BindIn in)
         {
             final String key = in.value();
 
-            return new Binder<Annotation, Iterable>()
+            return new Binder<Annotation, Object>()
             {
                 @Override
-                public void bind(SQLStatement q, Annotation bind, Iterable coll)
+                public void bind(final SQLStatement q, final Annotation bind, final Object arg)
                 {
+                    final Object[] arr = Util.toArray(arg);
+
                     int idx = 0;
-                    for (Object s : coll) {
+                    for (final Object s : arr)
+                    {
                         q.bind("__" + key + "_" + idx++, s);
                     }
                 }
             };
         }
     }
+
+    final class Util
+    {
+        private Util()
+        {
+        }
+
+        static Object[] toArray(final Object obj)
```

Comment:
Hmm, yeah, I suppose it does create a little less cruft/shortcuts a few steps... I mildly dislike the idea of passing iterators around because I prefer method arguments to be "natural" and obvious rather than obfuscated/indirect, but if I'm honest with myself that's not much of a good argument anyway. I'll patch it